firedrake/interpolation.py

```diff
@@ -124,31 +128,278 @@ class Interpolator(object):
+            # vom_dest_node_coords_in_m_src uses the parallel decomposition of
+            # the destination node coordinates in the source mesh. I can get
+            # back to the original ordering, which matches the parallel
+            # decomposition of the nodes of V_dest, by using the input_ordering
+            # attribute of the VOM
+            P0DG_vom = firedrake.FunctionSpace(
+                self.vom_dest_node_coords_in_m_src, "DG", 0
+            )
+            P0DG_vom_i_o = firedrake.FunctionSpace(
+                self.vom_dest_node_coords_in_m_src.input_ordering, "DG", 0
+            )
+            f_src_at_dest_node_coords_m_src_decomp = firedrake.Function(
+                P0DG_vom
+            ).interpolate(f_src)
+            f_src_at_dest_node_coords_m_dest_decomp = firedrake.Function(P0DG_vom_i_o)
+            if default_missing_val:
+                # By doing this here, the interpolation will leave these values
+                # (which aren't in the source mesh) unchanged.
+                f_src_at_dest_node_coords_m_dest_decomp.dat.data_wo[
+                    :
+                ] = default_missing_val
+            elif self._allow_missing_points:
+                # We might have points which are not in the source mesh which
+                # we will remove later.
+                f_src_at_dest_node_coords_m_dest_decomp.dat.data_wo[:] = numpy.nan
+            else:
+                pass  # default dat values are zero
+            f_src_at_dest_node_coords_m_dest_decomp.interpolate(
+                f_src_at_dest_node_coords_m_src_decomp
+            )
+
+            # we can now confidently assign this to a function on V_dest
+            if default_missing_val or not self._allow_missing_points:
+                # We've either already set a default to use or we know we wont
+                # have any missing points so it's safe to assign all values.
+                output.dat.data_wo[
+                    :
+                ] = f_src_at_dest_node_coords_m_dest_decomp.dat.data_ro[:]
+            else:
+                # Otherwise we need skip over the NaNs such that our output
+                # function values are left unaffected by the missing points.
+                indices = numpy.where(
+                    ~numpy.isnan(f_src_at_dest_node_coords_m_dest_decomp.dat.data_ro)
+                )[0]
+                output.dat.data_wo[
+                    indices
+                ] = f_src_at_dest_node_coords_m_dest_decomp.dat.data_ro[indices]
+
+        else:
+            # We would have to set up everything again if we wanted to do this
+            # properly, but we can cheat and interpolate the other way! This
+            # won't work when we enforce that transpose interpolators can only
+            # be used with cofunctions.
+            if not self.transpose_interpolator:
+                V_dest = self.expr.function_space()
+                self.transpose_interpolator = CrossMeshInterpolator(
```

Comment:
This is just interpolating the other way, which is NOT the same as getting the adjoint/transpose interpolator.